pkg/controllers/binding/common_test.go

```diff
@@ -379,3 +379,154 @@ func Test_shouldSuspendDispatching(t *testing.T) {
+				ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+					ReplicaSchedulingType: policyv1alpha1.ReplicaSchedulingTypeDivided,
+				},
+			},
+			want: false,
+		},
+		{
+			name:      "placement is nil",
+			replicas:  1,
+			placement: nil,
+			want:      false,
+		},
+		{
+			name:     "replica scheduling type is not divided",
+			replicas: 1,
+			placement: &policyv1alpha1.Placement{
+				ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+					ReplicaSchedulingType: policyv1alpha1.ReplicaSchedulingTypeDuplicated,
+				},
+			},
+			want: false,
+		},
+		{
+			name:     "replica scheduling type is divided",
+			replicas: 1,
+			placement: &policyv1alpha1.Placement{
+				ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+					ReplicaSchedulingType: policyv1alpha1.ReplicaSchedulingTypeDivided,
+				},
+			},
+			want: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := needReviseReplicas(tt.replicas, tt.placement); got != tt.want {
+				t.Errorf("needReviseReplicas() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_divideReplicasByJobCompletions(t *testing.T) {
+	tests := []struct {
+		name     string
+		workload *unstructured.Unstructured
+		clusters []workv1alpha2.TargetCluster
+		want     []workv1alpha2.TargetCluster
+		wantErr  bool
+	}{
+		{
+			name: "completions found",
+			workload: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"completions": int64(10),
+					},
+				},
+			},
+			clusters: []workv1alpha2.TargetCluster{
+				{Name: "cluster1", Replicas: 5},
+				{Name: "cluster2", Replicas: 5},
+			},
+			want: []workv1alpha2.TargetCluster{
+				{Name: "cluster1", Replicas: 5},
+				{Name: "cluster2", Replicas: 5},
+			},
+			wantErr: false,
+		},
+		{
+			name: "completions not found",
+			workload: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{},
+				},
+			},
+			clusters: []workv1alpha2.TargetCluster{
+				{Name: "cluster1", Replicas: 5},
+				{Name: "cluster2", Replicas: 5},
+			},
+			want:    []workv1alpha2.TargetCluster{},
+			wantErr: false,
+		},
+		{
+			name: "error in NestedInt64",
+			workload: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": "invalid",
+				},
+			},
+			clusters: []workv1alpha2.TargetCluster{
+				{Name: "cluster1", Replicas: 5},
+				{Name: "cluster2", Replicas: 5},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := divideReplicasByJobCompletions(tt.workload, tt.clusters)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("divideReplicasByJobCompletions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !compareTargetClusters(got, tt.want) {
+				t.Errorf("divideReplicasByJobCompletions() = %v, want %v", got, tt.want)
+			}
```

Comment:
Is it possible to sort the slices first and then use the `reflect.DeepEqual` method to determine?

Reply:
Sure...let me look into it

Reply:
Hey , @XiShanYongYe-Chang I have tried to make code more modular , we can try that but in future we are adding more tests which needs to do same operation , isn't this a good idead to make one separate helper function which can help use out for the same thing and adheres to design principle?

May be we can make one helper file like `common_helpers.go`. or something like that to put helper function used in whole tests , this will make codebase look more clear.

correct me if i am wrong !

Reply:
Sorry, I probably didn't express myself clearly. My mind is modified like this:

```diff
-                       if !compareTargetClusters(got, tt.want) {
+                       sort.Slice(got, func(i, j int) bool {
+                               return got[i].Name < got[j].Name
+                       })
+                       sort.Slice(tt.want, func(i, j int) bool {
+                               return tt.want[i].Name < tt.want[j].Name        
+                       })
+                       if !reflect.DeepEqual(got, tt.want) {
```

Reply:
Hey @XiShanYongYe-Chang  i have pushed the required changes , lets wait for ci workflows to pass after which we can review it right ?